### PR TITLE
Upgrade Next.js and don’t break e2e tests

### PR DESCRIPTION
**Pull request by flozia**
~Playwright tests are failing since PR #3501: Testing a couple of things to see if they pass in CI.~

We reverted PR #3501 which was upgrading to Next.js. This is another attempt at upgrading and not having the `e2e` tests fail. After the changes in this PR, I was not able to reproduce the Playwright failure anymore. For context: We are using a [non-standard NODE_ENV](https://nextjs.org/docs/messages/non-standard-node-env) and it seems like the previous version of Next.js did not make Playwrights web server stumble. Building and running the application without Playwright still works fine, though.

### CODE REVIEW

**Comment by rhelmer** [comment]
@flozia I reverted the whole commit 452a6827697a69bbee33690016ad3e585bbebe2f - if you want to use this to figure out exactly what's failing (and ideally fix it if it's Monitor app doing something wrong!) that would be welcome!

**Comment by flozia** [comment]
Test failures after upgrading Next.js solved in https://github.com/mozilla/blurts-server/pull/3500#issuecomment-1748954605.

**Comment by Vinnl** [comment]
@flozia That only fixes unit test failures - unfortunately I still don't know how to make the Playwright tests succeed :/

**Comment by flozia** [comment]
@Vinnl It seems like this fixing the issue (for now) and should unblock PR #3504.

File: .env-dist
Comment (pdehaan) [inline .env-dist:3]: Wait... why this change? Does this affect what tests run? Ah, maybe https://github.com/mozilla/blurts-server/blob/c0aa2eb9cf0ce41b74d1527eadf8a78c26977569/.github/workflows/playwright.yml#L51

Comment (flozia) [reply .env-dist:3]: Yes, that must be it! `NODE_ENV=development` is not a valid env variable when building Next.js with `npm run build`. This has been biting us in the past few days. Playwright only started failing on us after upgrading to`next: 13.5.4`. Building standalone without Playwright was still working.

**Review by Vinnl — APPROVED** [review]
Great job finding this Florian! As an alternative solution, I think we should probably remove the `NODE_ENV` value altogether - Next.js sets it to `development` for `npm run dev` and `production` for `npm run build` [by default](https://nextjs.org/docs/app/building-your-application/configuring/environment-variables#good-to-know):

> If the environment variable NODE_ENV is unassigned, Next.js automatically assigns development when running the next dev command, or production for all other commands.

By listing it in `.env-dist`, we might imply that it's something we should be setting ourselves, while in general, I think we just want to leave it to Next.js and other relevant tooling (e.g. Jest sets it to `test`).

**Comment by Vinnl** [comment]
https://www.youtube.com/watch?v=_ElWlUr_tw8

**Pull request merged**
